src/v2/keychains.js

```diff
@@ -99,6 +108,38 @@ Keychains.prototype.updatePassword = function(params, callback) {
   }).call(this).asCallback(callback);
 };
 
+/**
+ * Update the password used to decrypt a single keychain
+ * @param keychain - The keychain whose password should be updated
+ * @param oldPassword - The old password used for encrypting the key
+ * @param newPassword - The new password to be used for encrypting the key
+ * @param callback
+ * @returns {object}
+ */
+Keychains.prototype.updateSingleKeychainPassword = function({ keychain, oldPassword, newPassword }) {
+  if (!_.isString(oldPassword)) {
+    throw new Error('expected old password to be a string');
+  }
+
+  if (!_.isString(newPassword)) {
+    throw new Error('expected new password to be a string');
+  }
```

Comment:
More validation considerations:

1. What happens if both the old password + new password are the same?
1. What happens if the new password is an empty string?
1. What happens if the new password contains non-ascii characters that will fail during encryption?

Reply:
> More validation considerations:
> 
> 1. What happens if both the old password + new password are the same?

We will re-encrypt to the same password (essentially a no-op).

> 2. What happens if the new password is an empty string?

We will encrypt to the empty string.

> 3. What happens if the new password contains non-ascii characters that will fail during encryption?

I don't think non-ascii characters cause encryption failures, since the password is converted to a buffer fairly early in the encryption path. I can add a test to ensure this is the case.